lib/pure/includes/osenv.nim

```diff
@@ -160,38 +154,35 @@ else:
         else:
           proc strEnd(cstr: WideCString, c = 0'i32): WideCString {.
             importc: "wcschr", header: "<string.h>".}
+        var
```

Comment:
in this or future PR:
use a template to avoid repetition, eg:
```nim
template impl(fn1, fn2, n) =
        var
          env = fn1()
          e = env
        if e == nil: return # an error occurred
        while true:
          var eend = strEnd(e)
          let kv = $e
          var p = find(kv, '=')
          yield (substr(kv, 0, p-1), substr(kv, p+1))
          e = cast[cstring](cast[ByteAddress](eend)+n)
          if eend[1] == '\0': break
        discard fn2(env)
```
(adjust as needed)